drivers/storage/portworx/component/telemetry.go

```diff
@@ -806,16 +808,35 @@ func (t *telemetry) createCCMGoConfigMapTLSCertificate(
 func (t *telemetry) createCCMGoConfigMapTelemetryPhonehome(
 	cluster *corev1.StorageCluster,
 	ownerRef *metav1.OwnerReference,
-) error {
+) (bool, error) {
 	cloudSupportPort, _, _ := getCCMCloudSupportPorts(cluster, defaultPhonehomePort)
 	config, err := readConfigMapDataFromFile(configFileNameTelemetryPhonehome, map[string]string{
 		configParameterPortworxPort:         fmt.Sprint(getCCMListeningPort(cluster)),
 		configParameterRestCloudSupportPort: fmt.Sprint(cloudSupportPort),
 	})
 	if err != nil {
-		return err
+		return false, err
 	}
-	return k8sutil.CreateOrUpdateConfigMap(
+	data := map[string]string{
+		configFileNameTelemetryPhonehome: config,
+		TelemetryArcusLocationFilename:   getArcusTelemetryLocation(cluster),
+	}
+
+	dataChanged := true
+	existingConfigMap := &v1.ConfigMap{}
+	err = t.k8sClient.Get(
+		context.TODO(),
+		types.NamespacedName{
+			Name:      ConfigMapNameTelemetryPhonehome,
+			Namespace: cluster.Namespace,
+		},
+		existingConfigMap,
+	)
+	if err == nil && reflect.DeepEqual(existingConfigMap.Data, data) {
+		dataChanged = false
+	}
+
+	return dataChanged, k8sutil.CreateOrUpdateConfigMap(
```

Comment:
CreateOrUpdateConfigMap already calls get and has isModified flag, there is no need to make another get call in this function. you can return the flag from that function directly.